docs/variants/hardkernel_odroid_h4/recovery.md

```diff
@@ -0,0 +1,92 @@
+# Recovery
+
+## Intro
+
+The following documentation describes the process of recovering hardware from
+the brick state using an [RTE](../../transparent-validation/rte/introduction.md)
+and Dasharo open-source firmware.
+
+## Dual BIOS
+
+The **ODROID H4+ and ODROID-H4 Ultra** devices both have the Dual BIOS
+feature. If the device does not boot properly, the backup flash chip can be
+selected using the flash selector jumper.
+
+![](../../images/odroid_flash_switch.jpg)
+*First bios chip selected using the flash selector*
+
+1. Move the flash selector jumper to select the backup flash chip. If the
+   device does not boot from any of the flash chips, an external recovery
+   needs to be performed. In that case continue with the recovery instructions
+   for the ODROID H4 model.
```

Comment:
@philipandag please link that section I guess you mean "External flashing"

Reply:
Probably this https://wiki.odroid.com/odroid-h4/hardware/restore_h4_bios#using_tool_and_soldering_skills and https://wiki.odroid.com/odroid-h3/hardware/restore_h3_bios_usbprogrammer#with_usb_programmer_module_ch341a.
Had to test this recovery and it worked (flashed only one SPI, second one recovered with `flashrom -p internal`)

Reply:
> please link that section I guess you mean "External flashing"
Yes, the sections were renamed and I didn't update this sentence.

https://github.com/Dasharo/docs/pull/903/commits/aa4abb3b26367908a399f336b143dd20d13b4b27#diff-63d4fe5312d3d5c9d0e564e54adf0e43958113fb5d526eb698cd0ff31bd3dc79L19-L21

Do we want to give an URL to the instructions from wiki.odroid.com? If the point of this documentation is only to help the users maybe it would be better to only redirect them to the wiki instead of writing pretty much the same instructions again? I have added the URL to ODROID wiki instructions on flashing using the CH341A programmer as a second tab of the External Flashing section. https://github.com/Dasharo/docs/pull/903/commits/a3a7f52f0fc29e94767ca3a2bacd77341bdc19bf

In that case maybe an URL to the dual-bios recovery method should also be referenced? The wiki tells how to do it on Windows and using the EFI shell, that might be useful to some users.

Reply:
The documentation on odroid wiki includes instructions on flashing from UEFI shell and Windows which weren't tested by us, so I can't tell if they would require any additional information. Some weird flashing software which you need to launch from a USB drive is used there. Flashing using flashrom is not described on the wiki, so it does not overlap with what we have in our documentation, it may be included as an additional source.